shuup/campaigns/signal_handlers.py

```diff
@@ -5,17 +5,41 @@
 #
 # This source code is licensed under the AGPLv3 license found in the
 # LICENSE file in the root directory of this source tree.
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
 from shuup.campaigns.consts import (
     CAMPAIGNS_CACHE_NAMESPACE, CATALOG_FILTER_CACHE_NAMESPACE,
     CONTEXT_CONDITION_CACHE_NAMESPACE
 )
+from shuup.campaigns.models.basket_conditions import \
+    CategoryProductsBasketCondition
+from shuup.campaigns.models.basket_line_effects import \
+    DiscountFromCategoryProducts
+from shuup.campaigns.models.catalog_filters import CategoryFilter
 from shuup.campaigns.models.contact_group_sales_ranges import \
     ContactGroupSalesRange
 from shuup.core import cache
 
 from .utils import assign_to_group_based_on_sales
 
 
+def remove_category_from_campaigns(sender, category, **kwargs):
+    # raise an error if the category is part of an active campaign since this could result in a
+    # campaign that once applied to one category suddenly apply to the entire shop
+    if CategoryProductsBasketCondition.objects.filter(category=category, campaign__active=True).exists():
+        raise ValidationError(_("Category is used in the rules of an active basket campaign."))
+    if CategoryFilter.objects.filter(categories__id__exact=category.pk, campaign__active=True).exists():
+        raise ValidationError(_("Category is used in the rules of an active catalog campaign."))
+    for condition in CategoryProductsBasketCondition.objects.filter(category=category):
+        condition.delete()
```

Comment:
Don't delete the conditions, filters or effects just set the campaign is_active False. Maybe the soft delete is just temporary or something so let's not start deleting here.